src/UCTNode.cpp

```diff
@@ -322,13 +353,21 @@ UCTNode* UCTNode::uct_select_child(int color) {
         }
     }
 
-    auto numerator = static_cast<float>(std::sqrt((double)parentvisits));
+    if (parentvisits == 0) {
+        // sort by score.
+        parentvisits = 1;
+    }
+
+    auto numerator = std::sqrt((double)parentvisits);
     auto fpu_reduction = cfg_fpu_reduction * std::sqrt(total_visited_policy);
 
-    for (const auto& child : m_children) {
-        if (!child->valid()) {
-            continue;
-        }
+    // positive index mean expanded nodes, negative index for unexpanded.
+    auto best_index = 0;
+    auto best_value = -1000.0f;
+
+    for (auto i = size_t{0}; i < m_expanded.size(); i++) {
```

Comment:
Rather than the complexity of positive and negative indices, just initialize best_index to -1 and use that to mean it's a child of m_expanded, and use `for (auto child : m_expanded)` instead. It's simpler than changing the meaning of index.